File: exercises/D102/solution.py
value = input()
fare = float(value[1:])
ten_cents = round(fare * 10)
if ten_cents % 2 == 0:
    ten_cents_halved = ten_cents // 2
else:
    ten_cents_halved = ten_cents // 2 + 1
print(ten_cents_halved / 10.0)

File: exercises/D102/solution.java
import java.io.*;
import java.util.*;

class D102 {
  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);
    in.skip(".");
    double fare = in.nextDouble();
    int tenCents = (int)Math.round(fare * 10);
    int tenCentsHalved;
    if (tenCents % 2 == 0) {
        tenCentsHalved = tenCents / 2;
    } else {
        tenCentsHalved = tenCents / 2 + 1;
    }
    System.out.format("$%.1f", tenCentsHalved / 10.0);
  }
}
